extensions/functions_boolean.yaml

```diff
@@ -3,43 +3,110 @@
 scalar_functions:
   -
     name: or
-    description: The boolean or of two values using Kleene logic.
+    description: >
+      The boolean `or` using Kleene logic.
+
+      This function behaves as follows with nulls:
+
+          true or null = true
+
+          null or true = true
+
+          false or null = null
+
+          null or false = null
+
+          null or null = null
+
+      In other words, in this context a null value really means "unknown", and
+      an unknown value `or` true is always true.
+
+      Behavior for 0 or 1 inputs is as follows:
+        or() -> false
+        or(x) -> x
     impls:
       - args:
-          - value: boolean
+          - value: boolean?
```

Comment:
Note that the added question marks here and elsewhere also aren't necessary, since MIRROR behavior overrides nullability. But I'm fine with it either way.